core/src/main/scala/com/banno/kafka/package.scala

```diff
@@ -179,30 +120,133 @@ package object kafka {
         x.value == y.value
     }
 
-  implicit object ProducerRecordBifunctor extends Bifunctor[ProducerRecord] {
-    def bimap[A, B, C, D](
+  implicit object ProducerRecordBitraverse extends Bitraverse[ProducerRecord] {
+    override def bifoldLeft[A, B, C](fab: ProducerRecord[A, B], c: C)(
+        f: (C, A) => C,
+        g: (C, B) => C,
+    ): C =
+      g(f(c, fab.key), fab.value)
+
+    override def bifoldRight[A, B, C](fab: ProducerRecord[A, B], c: Eval[C])(
```

Comment:
This is even weirder to me, why is it all of a sudden important to introduce an `Eval[C]` type here? Why did we not have to do that for `bifoldLeft()`, which conceptually should do the same thing, just in the other direction.

Reply:
For posterity: answered this on a huddle.